### PR TITLE
Remove legacy Rails version conditionals

### PR DESCRIPTION
**Pull request by willnet**
The current sorcery now supports Rails >= 6.1 and Ruby >= 3.0. I've removed and cleaned up the code that was branching for older versions of Rails.

### CODE REVIEW

File: lib/generators/sorcery/install_generator.rb [hunk 1]
```diff
@@ -97,11 +97,7 @@ def only_submodules?
       end
 
       def migration_class_name
-        if Rails::VERSION::MAJOR >= 5
-          "ActiveRecord::Migration[#{Rails::VERSION::MAJOR}.#{Rails::VERSION::MINOR}]"
-        else
-          'ActiveRecord::Migration'
-        end
+        "ActiveRecord::Migration[#{Rails::VERSION::MAJOR}.#{Rails::VERSION::MINOR}]"
```
Comment (joshbuker) [inline lib/generators/sorcery/install_generator.rb:100]: I think we might be able to refactor this to just use `ActiveRecord::Migration::Current`, but non-blocking.

**Review by joshbuker — APPROVED** [review]
This is great, thanks @willnet!

File: lib/generators/sorcery/install_generator.rb (hunk 1 above)
Comment (willnet) [reply lib/generators/sorcery/install_generator.rb:100]: @joshbuker When using ActiveRecord::Migration::Current here, the content of the migration file might change when users upgrade their Rails version, so I think it's best to leave it as is.

Comment (joshbuker) [reply lib/generators/sorcery/install_generator.rb:100]: Wouldn't that also be true using this method?

**Pull request merged**